mapillary/utils/verify.py

```diff
@@ -218,17 +222,44 @@ def valid_id(id: int, image=True):
 
     # IF image == False, and error_check == True, this becomes True
     # IF image == True, and error_check == False, this becomes True
-    if image ^ EntityAdapter().is_image_id(id=id, fields=[]):
+    if image ^ is_image_id(id=id, fields=[]):
 
         # The EntityAdapter() sends a request to the server, checking
         # if the id is indeed an image_id, TRUE is so, else FALSE
 
         # Raises an exception of InvalidOptionError
         raise InvalidOptionError(
             param="id",
-            value=f"Id: {id}, image: {image}",
+            value=f"ID: {id}, image: {image}",
             options=[
-                "Id is image_id AND image is True",
-                "key is map_feature_id AND" "image is False",
+                "ID is image_id AND image is True",
+                "ID is map_feature_id AND image is False",
             ],
         )
+
+
+def is_image_id(id: int, fields: list = []) -> bool:
+    """Checks if the id is an image_id
+
+    :param id: The id to be checked
+    :type id: int
+
+    :param fields: The fields to be checked
+    :type fields: list
+
+    :return: True if the id is an image_id, else False
+    """
+    try:
+        res = requests.get(
+            Entities.get_image(
+                image_id=id,
+                fields=fields
+                if fields != []
+                else Entities.get_image_fields()
+            ),
+            headers={"Authorization": f"OAuth {Client.get_token()}"}
+        )
+        return res.status_code == 200
+        
+    except requests.HTTPError:
+        return False
```

Comment:
I'm not sure about this. I've been thinking of how the `EntityAdapter().is_image_id(id=id, fields=[])` and `is_image_id(id=id, fields=[])` and running the code in my head ... and both seem at most the same to me. Referring to this [function](https://github.com/facebookincubator/mapillary-python-sdk/blob/de2dc7c3e9fdb85ac2351e1363704f7d2103eb37/mapillary/models/api/entities.py#L192), both seem to do the same thing. 

The lines,

```python
        res = requests.get(
            Entities.get_image(
                image_id=id,
                fields=fields
                if fields != []
                else Entities.get_image_fields()
            ),
            headers={"Authorization": f"OAuth {Client.get_token()}"}
        )
```

Essentially perform the functionality implemented in '[EntityAdapter().fetch_image(...)](https://github.com/facebookincubator/mapillary-python-sdk/blob/de2dc7c3e9fdb85ac2351e1363704f7d2103eb37/mapillary/models/api/entities.py#L64)'.

I would like to know more about your thought process on this. Maybe there is a piece that I'm missing out on, but again for the most part `EntityAdapter().is_image_id(id=id, fields=[])` seems to do the same thing with same inputs and output sets.

Reply:
The only difference in this implementation is I used `requests.get()` to perform the test API call. I did this to avoid misleading logs from the `client` when providing a `map_feature` key. In the previous implementation, if you try to retrieve a map feature using its key, `valid_id` first checks if the key is an image key, it queries the API, which results in a `500` response from the server (because the key is not actually an image key), then the actual call to the `map_features` entity endpoint is made and the correct results are returned. This implementation avoids that by using `requests.get()` for the id validation check API call. Also, as this implementation uses `requests` and not the `Client` instantiation, I thought it wouldn't make sense to include it as a method in the `EntityAdapter` class. It'd better be added as a generalized validity checker. What do you think?

Screenshots for more clarification:
*Before the new implementation*
![image](https://user-images.githubusercontent.com/53264297/128270836-9d0d7c28-ce82-4bab-baea-c1d42da6fce0.png)

*After the new implementation*
![image](https://user-images.githubusercontent.com/53264297/128270976-000c699b-ba32-4414-939f-dc4171d3a780.png)